ref(ui): Get button styles from theme object

vuluongj20 opened the pull request:
Add a new property to the theme object containing common styles for form inputs and buttons, and update `<Button />` to use this new property.

**Before:**

<img width="926" alt="Screen Shot 2022-01-18 at 4 37 22 PM" src="https://user-images.githubusercontent.com/44172267/150041410-268b3703-65bb-46bf-856c-b60c7ea46422.png">

**After:**

<img width="926" alt="Screen Shot 2022-01-18 at 4 37 46 PM" src="https://user-images.githubusercontent.com/44172267/150041441-ac860ec7-ae54-4587-8e0f-0db1558419a4.png">

vuluongj20 committed:
ref(ui): Get button styles from theme object

 - Add a `form` property to the theme object containing common styles for form inputs and buttons.
 - Update `button.tsx` to use the newly-added `form` prop from `theme`

vuluongj20 commented on static/app/utils/theme.tsx:
```diff
@@ -677,6 +677,31 @@ const commonTheme = {
+      fontSize: '0.75rem',
+      lineHeight: '0.875rem',
+    },
+  },
```
In future PRs, I'll update form inputs to also use this property for sizing. This will ensure that buttons and inputs have consistent sizing.

vuluongj20 commented on static/app/utils/theme.tsx:
```diff
@@ -677,6 +677,31 @@ const commonTheme = {
+   */
+  form: {
+    default: {
+      height: '40px',
```
Explicitly setting a height here to fix the uneven sizing issue we've seen in a few places.

Before:
<img width="315" alt="Screen Shot 2022-01-18 at 4 42 58 PM" src="https://user-images.githubusercontent.com/44172267/150042019-59594372-e625-4658-acd4-d1ed5672e7d7.png">

After:
<img width="315" alt="Screen Shot 2022-01-18 at 4 43 16 PM" src="https://user-images.githubusercontent.com/44172267/150042055-7fc8ad18-3187-4ba7-9492-5bfe2d2b51f7.png">

priscilawebdev replied:
this issue was on my list for a while now... Thanks for fixing it @vuluongj20 👏

vuluongj20 commented on static/app/views/eventsV2/savedQuery/index.tsx:
```diff
@@ -457,10 +457,4 @@ const IconUpdate = styled('div')`
-    height: 38px;
-  }
-`;
-
```
This special style here was causing problems w/ the new button styles.

Before:
<img width="451" alt="AddToDashboardButton--b" src="https://user-images.githubusercontent.com/44172267/150042322-73f73331-d3ac-43b8-b8de-23325d2be41f.png">

After:
<img width="442" alt="Screen Shot 2022-01-18 at 4 46 53 PM" src="https://user-images.githubusercontent.com/44172267/150042412-5c3c842d-fc15-4489-b516-07fbf597ecf1.png">

vuluongj20 commented on static/app/views/organizationGroupDetails/actions/subscribeAction.tsx:
```diff
@@ -22,7 +22,7 @@ function SubscribeAction({disabled, group, onClick}: Props) {
       tooltipProps={{delay: 300}}
       priority={group.isSubscribed ? 'primary' : 'default'}
-      size="zero"
+      size="xsmall"
```
Not sure why this was set to "zero" to begin with.

Before:
<img width="191" alt="subscribeAction--b" src="https://user-images.githubusercontent.com/44172267/150042465-b477ef14-5420-4c88-87da-6c9250511bb7.png">

After:
<img width="191" alt="subscribeAction--a" src="https://user-images.githubusercontent.com/44172267/150042479-d2c7d80a-d315-483f-b04c-e070f8b90ec2.png">

vuluongj20 commented on static/app/utils/theme.tsx:
```diff
@@ -677,6 +677,31 @@ const commonTheme = {
+   * Common styles for form inputs & buttons, separated by size.
+   * Should be used to ensure consistent sizing among form elements.
+   */
+  form: {
+    default: {
+      height: '40px',
+      padding: '10px 16px',
+      fontSize: '0.875rem',
+      lineHeight: '1rem',
+    },
+    small: {
+      height: '34px',
+      padding: '8px 12px',
+      fontSize: '0.875rem',
+      lineHeight: '1rem',
+    },
+    xsmall: {
+      height: '28px',
+      padding: '6px 8px',
+      fontSize: '0.75rem',
+      lineHeight: '0.875rem',
+    },
```
In future PRs, I'll update form inputs to also use this property for sizing. This will ensure that buttons and inputs have consistent sizing.

vuluongj20 committed:
fix(test): Use 'Button' query for saved query test

Instead of the old 'AddToDashboardButton'

vuluongj20 committed:
fix(test): Add data-test-id to <SaveQueryButtonGroup />

vuluongj20 committed:
Merge branch 'master' into vuluong/button-refactor

priscilawebdev commented on static/app/components/button.tsx:
```diff
@@ -245,59 +233,36 @@ const StyledButton = styled(
   transition: background 0.1s, border 0.1s, box-shadow 0.1s;
 
+  ${p => p.priority === 'link' && `font-size: inherit; padding: 0;`}
+  ${p => p.size === 'zero' && `padding: 2px;`}
```
```suggestion
  ${p => p.size === 'zero' && `padding: ${space(0.25)}`}
```

priscilawebdev commented on static/app/utils/theme.tsx:
```diff
@@ -677,6 +677,31 @@ const commonTheme = {
+  form: {
+    default: {
+      height: '40px',
+      padding: '10px 16px',
```
not sure, but shouldn't we use the "spaces" defined in  https://github.com/getsentry/sentry/blob/master/static/app/styles/space.tsx ? 🤷‍♀️

vuluongj20 replied:
many of the sizes here are not available in `space`, e.g. `40px` and `10px`. It's possible that in the future we'll have button padding that strictly adheres to the base-8 sizes from `space`, at which time I'd use `space` here. For now, I'd keep these sizes for the sake of consistency with other components (esp. form inputs).

priscilawebdev commented on static/app/components/button.tsx:
```diff
@@ -18,7 +18,7 @@ type ButtonElement = HTMLButtonElement & HTMLAnchorElement & any;
 type Props = {
   priority?: 'default' | 'primary' | 'danger' | 'link' | 'success' | 'form';
-  size?: 'zero' | 'xsmall' | 'small';
+  size?: 'zero' | 'xsmall' | 'small' | 'default';
```
I'm not sure if we should provide the `default` type here, because if the user doesn't provide that prop, it will fall back to `default`, right? if we introduce the `default` type here, maybe some developer uses it explicitly

vuluongj20 replied:
ahh added this just to be complete, but you have a point. Just removed it!

priscilawebdev replied:
thanks!

vuluongj20 committed:
fix(ui): Make project page buttons smaller

vuluongj20 committed:
Update button.tsx

vuluongj20 committed:
style: Update comments for buttonPadding in theme.tsx

priscilawebdev reviewed (approved):
great work!

vuluongj20 committed:
fix(ui): Update button sizes in panel headers

With the new button sizes, `small` buttons look too large when used oin top of panels. `xsmall` would be a better option.

vuluongj20 committed:
Merge branch 'master' into vuluong/button-refactor

vuluongj20 committed:
fix(ui): Adjust button sizes & positions

Given the new button sizes, some buttons needed to be smaller/moved.

vuluongj20 committed:
fix(ui): Resize <TextCopyInput />'s copy button

vuluongj20 committed:
fix(ui): Recenter buttons in issue detail alerts

vuluongj20 committed:
fix(ui): Reduce size of <ErrorItem /> expand button

vuluongj20 committed:
Merge branch 'master' into vuluong/button-refactor
